hbase-server/src/main/java/org/apache/hadoop/hbase/master/cleaner/DirScanPool.java

```diff
@@ -100,8 +109,8 @@ synchronized void tryUpdatePoolSize(long timeout) {
       }
     }
     shutdownNow();
-    LOG.info("Update chore's pool size from {} to {}", pool.getParallelism(), size);
-    pool = new ForkJoinPool(size);
+    LOG.info("Update chore's pool size from {} to {}", pool.getPoolSize(), size);
+    pool = initializePool(size);
```

Comment:
For ThreadPoolExecutor, there is a setCorePoolSize method so we do not need to recreate it. I think we can make it as a final member.

Reply:
OK, let me address this in the new patch,  good suggestion.